iOS_SDK/OneSignalSDK/Source/OSStateSynchronizer.m

```diff
@@ -204,15 +198,13 @@ - (void)sendTagsWithAppId:(NSString *)appId
               sendingTags:(NSDictionary *)tags
               networkType:(NSNumber *)networkType
       processingCallbacks:(NSArray *)nowProcessingCallbacks {
-    let pushStateSyncronizer = [self getPushStateSynchronizer];
-    let emailStateSyncronizer = [self getEmailStateSynchronizer];
-    
+    let stateSyncronizer = [self getStateSynchronizers];
+    // Begin constructing the request for the external id update
     let requests = [NSMutableDictionary new];
-    requests[OS_PUSH] = [pushStateSyncronizer sendTagsWithUserId:_currentSubscriptionState.userId appId:appId sendingTags:tags networkType:networkType emailAuthHashToken:nil externalIdAuthHashToken:_currentSubscriptionState.externalIdAuthCode];
-    
-    if (emailStateSyncronizer)
-        requests[OS_EMAIL] = [emailStateSyncronizer sendTagsWithUserId:_currentEmailSubscriptionState.emailUserId appId:appId sendingTags:tags networkType:networkType emailAuthHashToken:_currentEmailSubscriptionState.emailAuthCode externalIdAuthHashToken:nil];
-    
+    for (OSUserStateSynchronizer* userStateSynchronizer in stateSyncronizer) {
+        requests[userStateSynchronizer.getChannelId] = [userStateSynchronizer sendTagsWithAppId:appId sendingTags:tags networkType:networkType];
+    }
+
     [OneSignalClient.sharedClient executeSimultaneousRequests:requests withSuccess:^(NSDictionary<NSString *, NSDictionary *> *results) {
```

Comment:
Recommend we address the duplication of these callbacks. As well it's extensibility of more channels. Example `NSDictionary *resultTags = results[OS_PUSH] ?: results[OS_EMAIL];`

Reply:
good catch!

Reply:
I made a method for parsing the results, I will love to have a more generic way of parsing each data from each channel, but that will take more work. I think like this can proceed, and in other phase improve more!

Reply:
I agree, we can clean this up more later